### DISCUSSION TITLE
A way to restrict to deploy raw manifests, allow only applications?

### DISCUSSION THREAD

**Discussion by iluwaa** [argoproj/argo-cd#15041] [Q&A] [answered]
Can I restrict to deploy raw manifests?

**Comment by crenshaw-dev** [comment]
Yes! You can disable Helm, Jsonnet, and Kustomize. That effectively restricts the users to just deploying raw manifets: https://argo-cd.readthedocs.io/en/latest/user-guide/tool_detection/#disable-built-in-tools

**Reply by iluwaa** [reply]
@crenshaw-dev  Thanks for answer! Sorry if I explain question wrong. I want to allow only helm deployments, restrict raw yamls, and allow some selected kinds of raw manifests.

**Reply by crenshaw-dev** [reply]
> restrict raw yamls, and allow some selected kinds of raw manifests.

Can you describe what kind of restrictions/allows you'd like to do?

**Reply by iluwaa** [reply]
@crenshaw-dev for example I want to allow kind: Ingress as raw manifest, but restrict kind: Certificate via raw, kind: Certificate can be applied only like part of helm chart.

**Reply by crenshaw-dev** [reply]
There's not really a way to do that today... you could open an enhancement request to enforce manifest type restrictions on a per-[project](https://argo-cd.readthedocs.io/en/latest/user-guide/projects/) basis.

**Reply by iluwaa** [reply]
@crenshaw-dev okay, thanks, last question, may be I can restrict all raw manifests and allow only to apply only helm charts?

**Reply by crenshaw-dev** [reply, marked as answer]
Unfortunately there's currently no toggle to disable raw manifests. It would be easy to implement though. Can you open an enhancement request for that as well?

**Reply by iluwaa** [reply]
sure, thanks
